parsec/core/types/backend_address.py

```diff
@@ -78,7 +84,24 @@ def from_url(cls, url: str):
         else:
             params = {}
 
-        path = unquote_plus(split.path)
+        if allow_http_redirection and split.scheme in ("http", "https"):
+            # `no_ssl` is defined by http/https scheme and shouldn't be
+            # overwritten by the query part of the url
+            params["no_ssl"] = ["true" if split.scheme == "http" else "false"]
```

Comment:
I'm a bit confused about this.. 

A `parsec` URL only needs to know about SSL if it plans to reach the backend, which is not the case for links to a file within a workspace. But we do need to use the proper protocol (`http` or `https`) if we want the backend to make the `http(s)` to `parsec` redirection. And we're assuming that the redirection and the backend API will always share the ssl/no_ssl config (since they run on same port using the same protocol), so we use the `http` or `https` information to set the `no_ssl` parameter in the backend address, possibly overwriting the existing one.

Am I getting this right?

Reply:
Yes, as you say this imply the http:// and parsec:// urls should use the same netloc (but this seems pretty resonable)

> possibly overwriting the existing one.

No, the `no_ssl` param is just not known. So we use the http vs https information to determine `no_ssl`. On an totally unrelated level, a `no_ssl` param could be provided as part of the url (because why not ? typically an attacker might want to mess with this...), but it shouldn't be taken into account.